blog/content/post/2020-07-21-secure-app-coding-practices.md

```diff
@@ -0,0 +1,74 @@
+
+We will try to approach the topic from a slightly easier side - what are the best practices for writing web applications?
+There are many of them, but here we'll cover some very basic rules to keep in mind.
+
+### Hide sensitive data
+
+This may seem obvious to you but the passwords for our accounts are so important that we should protect them as tightly as possible.
+If you make a test app and keep it in your repository you can say it doesn't matter. However, have you really not used this password somewhere? Hackers know very well that most internet users often use the exact same combination of characters. So you can expose yourself to the fact that someone will try to use the login data found, e.g. on Twitter or Google acount. An even worse scenario is possible, because what if you don't change your password before the app is released to the world? Even if you make your repo private from then on, there's no guarantee that someone hasn't already saved your data. Such neglect could cost you a lot. 
+First of all it is good practice to use separate channels of communication for sharing sensitive information such as a PIN or password. For example, you can use an HTTPS network connection to share encrypted data between the client and server, and then use APNS/GCM or SMS to share the PIN or token/key with the client app. That way, even if one channel of communication is compromised, the security of the overall system remains intact. Additionaly most local development and hosting environments now provide a mechanism for registering custom [environment variables](https://docs.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-3.1&tabs=windows/). 
+
+Don't forget though that keeping your server code on the repo, even without passwords, can be risky. If your script has any vulnerabilities, the hacker will be able to easily detect them, and then attack the application published, for example to download data from the database using it.
+
+
+### Strong password
+Hiding your password won't do anything if it's just weak. Remember that hackers often operate on powerful hardware that makes code-cracking applications extremely efficient. How can we protect ourselves against it? The attack called "by dictionary" tries the most popular combinations from words that already exist. That can be phrases or words from the dictionary.
```

Comment:
This paragraph, or a new one, could use a few words about 2FA, such as authenticators, tokens - maybe even physical keys (Yubikey, etc.)

Reply:
I would like to address the issue of injections, neutralizing threats, "broken authentication" and testing  in another article because the topic is really extensive. Here I only focused on the basic practices of secure coding

Reply:
too patronizing: leave "remember that"
dictionary appears twice nearby